lib/compiler/src/compile.erl

```diff
@@ -923,6 +925,7 @@ asm_passes() ->
 
 binary_passes() ->
     [{iff,'to_dis',?pass(to_dis)},
+     {iff,beam_docs,?pass(beam_docs)},
```

Comment:
I think it would be better to place this pass in the `abstr_passes` just after `save_abstract_code` and for it to work in a similar way. That is extract the things that should be placed in the doc chunk and place it in the #compile record until emitted by the `binary` pass.